Source/WebCore/platform/graphics/cg/GraphicsContextCG.h

```diff
@@ -127,6 +129,15 @@ class WEBCORE_EXPORT GraphicsContextCG : public GraphicsContext {
 
     virtual FloatRect roundToDevicePixels(const FloatRect&, RoundingMode = RoundAllSides) const;
 
+    // Returns the platform context for draws.
+    CGContextRef contextForDraw();
```

Comment:
Maybe `contextForDrawing()`

It's going to be easy to make a mistake in future and get the context the wrong way. How about passing a flag to `platformContext()`, so we have to force people to choose a flag at all call sites?

Reply:
> Maybe `contextForDrawing()`

Done.

> It's going to be easy to make a mistake in future and get the context the wrong way. How about passing a flag to `platformContext()`, so we have to force people to choose a flag at all call sites?

I think with current code it's actually very *hard* to make a mistake. `platformContext()` will use the the correct function, e.g. `contextForDrawing()`. So it's only a very slight perf problem. Currently there are no clients outside GraphicsContextGC that would be affected, and probably won't be.  Future commits will remove platformContext() usage from GraphicsContextGC. Future future commits, hopefully, will remove platformContext() usage from the rest of CG codepaths.

The function `GraphicsContext::platformContext()` is a bug in and of itself. I want to work on removing it instead of modifying, as it is error prone and needs redundancy in form of `hasPlatformContext()`.

Contrasting:

```
if (!c.hasPlatformContext())
   return;
// Hopefully this is CG-based??
c.platformContext(ForDrawing);
c.platformContext(ForState);
```
vs.
```
if (!is<GraphicsContextCG>(c))
   return;
auto context = downcast<GraphicsContextCG>(c);
context.contextForDrawing();
context.contextForState();
```
I think the latter is the sane approach and the direction we should put the effort in.

In other words:
 - platformContext() function is a bug, since  not all GraphicsContexts have platformContext() or there's no single universal platformContext() for a project. Hence it does not make sense to have such a function.
 - I already added the correct way to access the context: two *different* functions for different purposes. It is better than one vague function that turns into two distinct functions based on the two-value argument being passed. If the currently proposed approach wasn't better, it would get a review comment of form "would you merge your `contextForState` and `contextForDrawing` into one function with a parameter `contextFor(Case)`. I willing to entertain the thought if you think this is the way. However, also in this the correct solution is not to overload the old platformContext() that is opaque function across all the `GraphicsContexts` -- there are `GraphicsContexts` that do not need the `Case` parameter.

Reply:
You still have `contextForDraw()` in this PR.